core/domain/summary_services.py

```diff
@@ -352,29 +355,68 @@ def _generate_query(categories):
     return results
 
 
-def get_featured_exploration_summary_dicts(language_codes):
-    """Returns a list of featured explorations with the given language code.
-
-    The return value is sorted in decreasing order of search rank.
+def require_activity_ids_to_be_public(activity_ids):
+    """Raises an exception if any activity id in the list does not exist, or
+    is not public, or if the list has duplicates.
     """
-    filtered_exp_summaries = [
-        exp_summary for exp_summary in
-        exp_services.get_featured_exploration_summaries().values()
-        if exp_summary.language_code in language_codes]
+    exploration_ids, collection_ids = utils.split_activity_ids_by_type(
+        activity_ids)
+
+    exp_summaries = exp_services.get_exploration_summaries_matching_ids(
+        exploration_ids)
+    col_summaries = collection_services.get_collection_summaries_matching_ids(
+        collection_ids)
+
+    for index, summary in enumerate(exp_summaries):
+        if summary is None:
+            raise Exception(
+                'Cannot feature non-existent exploration with id %s' %
+                exploration_ids[index])
+        if summary.status == rights_manager.ACTIVITY_STATUS_PRIVATE:
+            raise Exception(
+                'Cannot feature private exploration with id %s' % summary.id)
+
+    for index, summary in enumerate(col_summaries):
```

Comment:
I feel like this function can have a lot of de-duplication by consolidating the similarities between checking for private activities and collections.